django/middleware/csrf.py

```diff
@@ -105,19 +105,38 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
+    def _get_current_token(self, request):
+        try:
+            if settings.CSRF_USE_SESSIONS:
+                return _sanitize_token(request.session[settings.CSRF_COOKIE_NAME])
+            else:
+                return _sanitize_token(request.COOKIES[settings.CSRF_COOKIE_NAME])
+        except KeyError:
+            return None
+
+    def _set_token(self, request, response):
+        if settings.CSRF_USE_SESSIONS:
+            request.session[settings.CSRF_COOKIE_NAME] = request.META["CSRF_COOKIE"]
```

Comment:
I agree, but I did not want change this, because while it is not a documented API, I'm almost sure that someone out there uses this META entry variable to e.g. supply the key to JavaScrpt.